### ISSUE TITLE
Tractor drives in circles

### ISSUE THREAD

**Issue by Flashsolutions** [Stephan-S/FS19_AutoDrive#856]
AutoDrive Version: 1.0.6.8 (note, the revision number was not updated from 1.0.6.7)
Release or custom zip: Release
Map (if applicable): Eiserholt
Vehicle / type (applicable):
Massey Ferguson 7726 / Krampe Bandit 750 trailer 

**Describe the bug**
#1. The version number for the release was not updated from 1.0.6.7 (I downloaded twice to be sure)
#2. While using AutoDrive in combine unload mode, the tractor got stuck driving in circles and would not find the path to the combine.

**If applicable - Steps to reproduce the behavior**
Do not know how to recreate... just happened once while attempting to unload the harvestor but I took a video of it...

**Screenshots**
https://youtu.be/r99f7B87E1E

**Comment by Iwan1803** [comment]
#1 was fixed in the custom zip, but not in the ZIP release
#2 is known, but at the moment it doesn't exist a fix for this

**Comment by geo-luchs** [comment]
Hi, Not Sure if ist's important vor the path findet, gut sich Option in AD of unloader did you Setfor Fruit Saving?
Yes or no?
Could it be possible,  that AD tries to Save the Fruit, (even he is driving through), but the tresher would  have the pipe in die Fruit?
Maybe try to activate   Fruit Saving vor dresher, too.

**Comment by Iwan1803** [comment]
Problem is, that we can't see the white line which the uploader would like to drive on.
It sounds like the problem i've got sometimes. The uploader tries to drive to the tresher but then it drives in circles around the start point of the white line.

**Comment by raxtor** [comment]
https://github.com/Stephan-S/FS19_AutoDrive/issues/542#issuecomment-552126645
Here is hint how to use console command for better debugging

**Comment by Flashsolutions** [comment]
@geo-luchs The fruit saving was set to NO.

**Comment by Stephan-S** [comment]
New console command is now "ADsetDebugChannel" @raxtor 
I changed this when I moved the debug functions to allow for channels rather than levels.

**Comment by Flashsolutions** [comment]
Should ADsetDebugChannel be left on or does it interfere with normal operations?

**Comment by Stephan-S** [comment]
it only adds debug print messages and infos.
If you enable the edit mode in the unloader, you will be able to see the result of the Pathfinder.

**Comment by Flashsolutions** [comment]
How do you reset it?  I get error messages if I add true or false to the command.

2019-11-11 15:31 dataS2/character/crow/crow.i3d (10.71 ms)
2019-11-11 15:39 Warning (script): 'bitAND': Argument 2 has wrong type. Expected: UInt. Actual: Nil 
2019-11-11 15:39   ..\..\..\src\base\scripting\lua\LuaScriptSystem.cpp (882): expectedType == Value::VoidType || expectedType == Value::ClassType
2019-11-11 15:39 LUA call stack:
2019-11-11 15:39   =C:/Users/UserName/Documents/My Games/FarmingSimulator2019/EIERSHOLT PLUS/FS19_AutoDrive/scripts/AutoDriveUtilFuncs.lua (211) : bitAND
2019-11-11 15:39 Error: Running LUA method 'setDebugChannel'.
2019-11-11 15:39 C:/Users/UserName/Documents/My Games/FarmingSimulator2019/EIERSHOLT PLUS/FS19_AutoDrive/scripts/AutoDriveUtilFuncs.lua:212: attempt to perform arithmetic on local 'newDebugChannel' (a nil value)
2019-11-11 15:41 Error: No depth map available, but shape 'drivingDust' (emitter1Ref) with material 'fx_smoke01_mat' uses a custom shader which requires reading depth data.
2019-11-11 15:42 data/shared/ai/trafficCollision.i3d (0.87 ms)
2019-11-11 15:45 CourseGeneratorScreen:onCreate()
2019-11-11 15:46 CourseGeneratorScreen:onCreate()
2019-11-11 15:47  callDriverToCombine
2019-11-11 15:47 7726:  startPathPlanningToCombine - minTurnRadius: 7.5 AI: 10 tools: 0
2019-11-11 15:49 Warning (script): 'bitAND': Argument 2 has wrong type. Expected: UInt. Actual: Nil 
2019-11-11 15:49   ..\..\..\src\base\scripting\lua\LuaScriptSystem.cpp (882): expectedType == Value::VoidType || expectedType == Value::ClassType
2019-11-11 15:49 LUA call stack:
2019-11-11 15:49   =C:/Users/UserName/Documents/My Games/FarmingSimulator2019/EIERSHOLT PLUS/FS19_AutoDrive/scripts/AutoDriveUtilFuncs.lua (211) : bitAND
2019-11-11 15:49 Error: Running LUA method 'setDebugChannel'.
2019-11-11 15:49 C:/Users/UserName/Documents/My Games/FarmingSimulator2019/EIERSHOLT PLUS/FS19_AutoDrive/scripts/AutoDriveUtilFuncs.lua:212: attempt to perform arithmetic on local 'newDebugChannel' (a nil value)
2019-11-11 15:51 dataS2/character/humans/player/player02.i3d (239.35 ms)
2019-11-11 15:53 CourseGeneratorScreen:onCreate()

**Comment by geo-luchs** [comment]
Btw. Also CP is sometimes driving crazy Circles, even  the calculated Shown path looks very simple and  definitly has no conflict with turn radius  of the vehicles.
It's Not only an AD Problem. Unfortunalty AD is Not Coming out of the Circles.

**Comment by Stephan-S** [comment]
ADsetDebugChannel 0

**Comment by Flashsolutions** [comment]
So, to be clear, ADsetDebugChannel 0 and 1 ?  Or are there different channel numbers and if so what do they imply.  Is there a document anywhere to help us know these things?  I know CP has different channel numbers for debug.  Also, I have tried to figure out how to prevent CP from backing up the combine and swerving to the right to make room for the tractor which drives up, unload the combine, and then the combine turns back to the left and drives into the tractor as is trying to back out of the way.  I realize that is not an AutoDrive question, but maybe someone knows the answer to that question?

**Comment by Flashsolutions** [comment]
Okay, I managed to make it do it again.  It is doing it right now.  I have paused the game.  Tell me what I provide to help.

**Comment by Flashsolutions** [comment]
![Farming Simulator 19 11_11_2019 4_46_54 PM](https://user-images.githubusercontent.com/60280/68623700-16fbc300-04a3-11ea-8d6b-398642288ff2.png)

**Comment by Flashsolutions** [comment]
Sorry, did not mean to close the issue.

**Comment by Flashsolutions** [comment]
Maybe this will help.... I slowed the speed of the tractor down to 11 mph and after a minute or so, the tractor left the circle and headed to the combine...

**Comment by raxtor** [comment]
Problem is that tractor has calculated radius which is smaller than radius with implement/trailer... it leads to circle stuck cuz tractor is not able catch calculated target point. Just my opinion... 
I proposed already if there is error msg for stuck than try recorvery maneuever like neutral steering for a few meters and than try again find new path.

**Comment by Flashsolutions** [comment]
I did not see a stuck error message in either occurrence.  If I remember correctly, CP has a turn radius parameter that can be adjusted but guess that would not be applicable to AD.

**Comment by Flashsolutions** [comment]
Perhaps, maybe a loop counter of the number of times the tractor has made the same path and then gradually slow the speed down or set the speed to a fixed slower value to see if it can recover?  I'm not familiar with the implementation to know if even feasible but I have seen AD overshoot turns on the road paths due to driving too fast.  I think the tractor was set to 20 when it got into trouble.  11 is okay for the field, but a bit slow once on the road to the grain sell point.

**Comment by raxtor** [comment]
or just reduce turn radius about some value, lets say 2/3 of actual..
error msg should be shown after a few circles, not immediately...

**Comment by Flashsolutions** [comment]
![Farming Simulator 19 11_11_2019 5_48_41 PM](https://user-images.githubusercontent.com/60280/68627376-9d1c0780-04ab-11ea-8ffe-fe27a26789d5.png)
Dang!  Now the tractor has run directly into the front of the combine! Maybe the slower speed caused this?

**Comment by raxtor** [comment]
@Flashsolutions btw which map is that?

**Comment by Flashsolutions** [comment]
This is the Eiersholt map.

**Comment by Flashsolutions** [comment]
None of the fields are rectangle like Ravenport and some others.  Every field is odd shaped.

**Comment by raxtor** [comment]
thx, I like it, I play County Lines 2.0, its 4x map similar to your

**Comment by Flashsolutions** [comment]
I loaded the latest Beta yesterday which appeared to have a fix to slow the tractor.  It seems to be working now.  I ran the same course several times and observed the tractor getting stuck, then slowing down and ramping back up and the tractor was able to resume.  I am going to close it for now.  I will reopen a new issue if I find it still occurring, but I drove a couple of field and even added a second tractor on the same field and did not see a reoccurrence.
